Source/NSCollectionView.m

```diff
@@ -1114,4 +1500,556 @@ - (BOOL) wantsPeriodicDraggingUpdates
+{
+  return nil;
+}
+
+- (void) scrollToItemsAtIndexPaths: (NSSet *)indexPaths 
+                    scrollPosition: (NSCollectionViewScrollPosition)scrollPosition
+{
+}
+
+/* Creating Collection view Items */
+
+- (NSCollectionViewItem *) makeItemWithIdentifier: (NSUserInterfaceItemIdentifier)identifier 
+                                     forIndexPath: (NSIndexPath *)indexPath
+{
+  NSCollectionViewItem *item =
+    [_dataSource collectionView: self itemForRepresentedObjectAtIndexPath: indexPath];
+  NSNib *nib = [self _nibForClass: [item class]];
+  BOOL loaded = [nib instantiateWithOwner: item
+                          topLevelObjects: NULL];
+
+  if (loaded == NO)
+    {
+      item = nil;
+      NSLog(@"Could not load model %@", nib);
+    }
+  
+  return item;
+}
+
+- (NSView *) makeSupplementaryViewOfKind: (NSCollectionViewSupplementaryElementKind)elementKind 
+                          withIdentifier: (NSUserInterfaceItemIdentifier)identifier 
+                            forIndexPath: (NSIndexPath *)indexPath
+{
+  return nil;
+}
+
+- (void) registerClass: (Class)itemClass 
+ forItemWithIdentifier: (NSUserInterfaceItemIdentifier)identifier
+{
+  [self registerClass: itemClass
+        forSupplementaryViewOfKind: GSNoSupplementaryElement
+        withIdentifier: identifier];
+}
+
+- (void) registerNib: (NSNib *)nib 
+         forItemWithIdentifier: (NSUserInterfaceItemIdentifier)identifier
+{
+  [self registerNib: nib
+        forSupplementaryViewOfKind: GSNoSupplementaryElement
+        withIdentifier: identifier];
+}
+
+- (void) registerClass: (Class)viewClass 
+         forSupplementaryViewOfKind: (NSCollectionViewSupplementaryElementKind)kind 
+         withIdentifier:(NSUserInterfaceItemIdentifier)identifier
+{
+  NSMapTable *t = nil;
+
+  t = [_registeredClasses objectForKey: kind];
+  if (t == nil)
+    {
+      t = [NSMapTable weakToWeakObjectsMapTable];
+      [_registeredClasses setObject: t
+                             forKey: kind];
+    }
+
+  [t setObject: viewClass forKey: identifier];
+}
+
+- (void) registerNib: (NSNib *)nib 
+         forSupplementaryViewOfKind: (NSCollectionViewSupplementaryElementKind)kind 
+      withIdentifier: (NSUserInterfaceItemIdentifier)identifier
+{
+  NSMapTable *t = nil;
+  
+  t = [_registeredNibs objectForKey: kind];
+  if (t == nil)
+    {
+      t = [NSMapTable weakToWeakObjectsMapTable];
+      [_registeredNibs setObject: t
+                          forKey: kind];
+    }
+
+  [t setObject: nib forKey: identifier];
+}
+
+/* Providing the collection view's data */
+
+- (id<NSCollectionViewDataSource>) dataSource
+{
+  return _dataSource;
+}
+
+- (void) setDataSource: (id<NSCollectionViewDataSource>)dataSource
+{
+  _dataSource = dataSource;
+  [self reloadData];
+}
+
+/* Configuring the Collection view */
+
+- (NSNib *) _nibForClass: (Class)cls
+{
+  NSString *clsName = NSStringFromClass(cls);
+  NSNib *nib = [[NSNib alloc] initWithNibNamed: clsName
```

Comment:
Looks like this leaks the NSNib object

Reply:
Why did you resolve this? The NSNib object still does not get autoreleased here.